### PR TITLE
[#934] suppress deprecated warnings from the api server as we know we…

### PR DESCRIPTION
**Pull request by gtully**
… need to work with deprecated crs

### CODE REVIEW

**Review by gaohoward — APPROVED** [review]

**Comment by gaohoward** [comment]
The downside of this is that there are no warnings at all....

**Comment by gtully** [comment]
> The downside of this is that there are no warnings at all....

only in the operator logs, for the client that the operator uses to chat to the api server.

end users of kubectl will have their own client will see the warning in the normal way.

Also for the kube client in the test suite. That will still process the warnings as per the defaults.

File: main.go [hunk 1]
```diff
@@ -184,6 +185,9 @@ func main() {
 		Logger:                 setupLog,
 	}
 
+	mgrOptions.Client.WarningHandler.SuppressWarnings = true
+	rest.SetDefaultWarningHandler(&rest.NoWarnings{})
```
Comment (brusdev) [inline main.go:189]: Could those warnings be logged with the debug level by creating a custom WarningHandler?

**Review by brusdev — COMMENTED** [review]
The WarningHandler is a great solution!

File: main.go (hunk 1 above)
Comment (gtully) [reply main.go:189]: They could, and maybe in the future there are other warnings that may be all we get. let me change that to debug logging.

**Comment by gtully** [comment]
- --zap-log-level=2

`2024-05-29T09:52:42.665Z LEVEL(-2) The ActiveMQArtemisAddress CRD is deprecated. Use the spec.brokerProperties attribute in the ActiveMQArtemis CR to create addresses and queues instead {"code": 299, "agent": "-"}
`

**Comment by gtully** [comment]
> > The downside of this is that there are no warnings at all....
> 
> only in the operator logs, for the client that the operator uses to chat to the api server.
> 
> end users of kubectl will have their own client will see the warning in the normal way.
> 
> Also for the kube client in the test suite. That will still process the warnings as per the defaults.

it makes sense to log at level 2, thanks for the review

**Review by brusdev — APPROVED** [review]

**Pull request merged**